src/scripts/archive_dataset/archive_dataset.py

```diff
@@ -0,0 +1,141 @@
+                        help="The AWS profile to use.")
+    args = parser.parse_args()
+    return args
+
+
+def get_source_and_dest_prefix(s3_client, bucket, app, dataset, dataset_version):
+    source_and_dest = dict()
+    study_prefix_obj = s3_client.list_objects_v2(
+            Bucket=bucket,
+            Prefix=f"{app}/",
+            Delimiter="/")
+    study_prefixes = [cp["Prefix"] for cp in study_prefix_obj["CommonPrefixes"]]
+    for study_prefix in study_prefixes:
+        parquet_prefix = f"{study_prefix}parquet/"
+        parquet_dataset_prefix_obj = s3_client.list_objects_v2(
+                Bucket=bucket,
+                Prefix=parquet_prefix,
+                Delimiter="/")
+        descendent_dataset_prefixes = [
+                cp["Prefix"] for cp in parquet_dataset_prefix_obj["CommonPrefixes"]
+                if f"{dataset}_" in cp["Prefix"] and dataset_version in cp["Prefix"]]
+        latest_archive_dataset_update = get_archive_dataset_update_number(
+                s3_client=s3_client,
+                bucket=bucket,
+                study_prefix=study_prefix,
+                dataset=dataset,
+                dataset_version=dataset_version)
+        for descendent_dataset_prefix in descendent_dataset_prefixes:
+            source_path = f"s3://{bucket}/{descendent_dataset_prefix}"
+            descendent_dataset = "_".join(
+                    descendent_dataset_prefix
+                    .split("/")[-2]
+                    .split("_")[:-1])
+            dest_path = (f"s3://{bucket}/{study_prefix}parquet/"
+                         f"archive/{descendent_dataset}_{dataset_version}_"
+                         f"{latest_archive_dataset_update+1}/")
+            source_and_dest[source_path] = dest_path
+    return source_and_dest
+
+
+def get_archive_dataset_update_number(s3_client, bucket, study_prefix,
+                                      dataset, dataset_version):
+    archive_prefix = f"{study_prefix}parquet/archive/"
+    archive_dataset_prefix_obj = s3_client.list_objects_v2(
+            Bucket=bucket,
+            Prefix=archive_prefix,
+            Delimiter="/")
+    relevant_archive_dataset_prefixes = []
+    if "CommonPrefixes" in archive_dataset_prefix_obj.keys():
+        relevant_archive_dataset_prefixes = [
+                cp["Prefix"] for cp in archive_dataset_prefix_obj["CommonPrefixes"]
+                if f"{dataset}_{dataset_version}" in cp["Prefix"]]
+    if len(relevant_archive_dataset_prefixes) == 0:
+        # No datasets of this type and version in archive
+        return 0
+    preexisting_update_nums = [
+            prefix.split("_")[-1][:-1] for prefix in relevant_archive_dataset_prefixes]
+    latest_version = max([int(n) for n in preexisting_update_nums])
+    return latest_version
+
+
+def copy_source_to_dest(source_and_dest):
+    for source in source_and_dest:
+        dest = source_and_dest[source]
+        bash_command = f"aws s3 cp --recursive {source} {dest}"
```

Comment:
Why not use `subprocess.check_call`?  Then you can use a list instead of f string interpolation

Reply:
`subprocess.run` was introduced in python3.5 to replace `subprocess.check_*` like methods. Both methods take a list.

Reply:
Ahh, I'm behind times a bit - thanks for letting me know!